pages/calender.jsx

```diff
@@ -1,114 +1,62 @@
 import { Link } from "react-router-dom";
+import { useState, useRef, useEffect } from "react";
+import TaskList from "../components/calendar_components/TaskList";
+import { v4 as uuid } from "uuid";
 
 const ToDoList = () => {
+  const [tasks, setTasks] = useState([]);
+  const taskNameRef = useRef();
+  const LOCAL_STORAGE_KEY = "taskApp.tasks";
+
+  useEffect(() => {
+    const storedTasks = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
+    if (storedTasks) setTasks(storedTasks);
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(tasks));
+  }, [tasks]);
+
+  function toggleTask(id) {
+    const newTasks = [...tasks];
+    const task = newTasks.find((task) => task.id === id);
+    task.complete = !task.complete;
+    setTasks(newTasks);
```

Comment:
Very good making a copy first and then doing the `find()` method.

Another solution would be: (But just FYI) 
```js
const newTask = tasks.map(task => {
    if (task.id === id) {
        task.complete = !task.complet
        return task 
    } else {
        return task 
    }
})
```